### DISCUSSION TITLE
How to define function type by its argument name

### DISCUSSION THREAD

**Discussion by STofone** [python/typing#1903] [General]
```python
func_list = []
def run_func_list(arg1, arg2):
    for func in func_list:
        func(arg1=arg1, arg2=arg2)

def func1(arg1, arg2):
    pass
func_list.append(func1)

def func1(arg2, arg1):
    pass
func_list.append(func1)

def func3(**kwargs):
    pass
func_list.append(func3)
```
I have a list that will be traversed and call function in it by named argument.
It supports multiple parameter forms of function.
How do i annotate `func_list` to tell others what it supports

**Comment by hauntsaninja** [comment]
Use a callback protocol: https://mypy.readthedocs.io/en/stable/protocols.html#callback-protocols
